src/librustc/ty/codec.rs

```diff
@@ -215,6 +215,20 @@ where
     Ok(tcx.mk_substs((0..len).map(|_| Decodable::decode(decoder)))?)
 }
 
+#[inline]
+pub fn decode_place<D>(decoder: &mut D) -> Result<mir::Place<'tcx>, D::Error>
+where
+    D: TyDecoder<'tcx>,
+{
+    let base: mir::PlaceBase<'tcx> = Decodable::decode(decoder)?;
+    let len = decoder.read_usize()?;
+    let interned: Vec<mir::PlaceElem<'tcx>> =
```

Comment:
are there no nice wrappers out there for (de)serializign a `List`?

Reply:
I've fixed the deserialization and made it look a bit better. But what we have is very similar to what's done with ty https://github.com/rust-lang/rust/blob/e0fe4846e5ab3f5d8810280bbe0d217cafd491c9/src/librustc/ty/codec.rs#L247